samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -136,6 +137,44 @@ public synchronized String getEphemeralPath() {
     return children;
   }
 
+  /**
+   * Method is used to read processors PIDs
```

Comment:
It is not very obvious if you are referring to the OS's Process ID as PID or the Processor's ID as PID. Please change "PID" to explicitly read "processorId" and the id assigned by zk can be referred as znode id. It will be useful to use the same terminology across the codebase.